packages/datadog-instrumentations/test/mysql2.spec.js

```diff
@@ -0,0 +1,718 @@
+'use strict'
+
+const { channel } = require('../src/helpers/instrument')
+const agent = require('../../dd-trace/test/plugins/agent')
+const { assert } = require('chai')
+const semver = require('semver')
+
+describe('mysql2 instrumentation', () => {
+  withVersions('mysql2', 'mysql2', version => {
+    function abort ({ sql, abortController }) {
+      assert.isString(sql)
+      const error = new Error('Test')
+      abortController.abort(error)
+
+      if (!abortController.signal.reason) {
+        abortController.signal.reason = error
+      }
+    }
+
+    function noop () {}
+
+    const config = {
+      host: '127.0.0.1',
+      user: 'root',
+      database: 'db'
+    }
+
+    const sql = 'SELECT 1'
+    let startCh, mysql2, shouldEmitEndAfterQueryAbort
+    let apmQueryStartChannel, apmQueryStart, mysql2Version
+
+    before(() => {
+      startCh = channel('datadog:mysql2:outerquery:start')
+      return agent.load(['mysql2'])
+    })
+
+    before(() => {
+      const mysql2Require = require(`../../../versions/mysql2@${version}`)
+      mysql2Version = mysql2Require.version()
+      // in v1.3.3 CommandQuery started to emit 'end' after 'error' event
+      shouldEmitEndAfterQueryAbort = semver.intersects(mysql2Version, '>=1.3.3')
+      mysql2 = mysql2Require.get()
+      apmQueryStartChannel = channel('apm:mysql2:query:start')
+    })
+
+    beforeEach(() => {
+      apmQueryStart = sinon.stub()
+      apmQueryStartChannel.subscribe(apmQueryStart)
+    })
+
+    afterEach(() => {
+      if (startCh?.hasSubscribers) {
+        startCh.unsubscribe(abort)
+        startCh.unsubscribe(noop)
+      }
+      apmQueryStartChannel.unsubscribe(apmQueryStart)
+    })
+
+    describe('lib/connection.js', () => {
+      let connection
+
+      beforeEach(() => {
+        connection = mysql2.createConnection(config)
+
+        connection.connect()
+      })
+
+      afterEach((done) => {
+        connection.end(() => done())
+      })
+
+      describe('Connection.prototype.query', () => {
+        describe('with string as query', () => {
+          describe('with callback', () => {
+            it('should abort the query on abortController.abort()', (done) => {
+              startCh.subscribe(abort)
+              const query = connection.query(sql, (err, _) => {
```

Comment:
Nit: Remove unused `_` parameter.

Applies for several cases along the code in this file.